src/plane.ts

```diff
@@ -9,19 +9,104 @@
 // -- Remove boxMesh and coneMesh
 
 import * as THREE from 'three';
+import {option} from 'fp-ts';
+import * as defaultPlane from './default-plane.json';
+
+interface basicGeometry {
+  wingspan: number;
+  length: number;
+  chord: number;
+}
+
+enum toControls {
+  pitch,
+  yaw,
+  roll,
+}
+
+interface controlLoop {
+  frequency: number;
+  interrupts: Array<toControls>;
+}
+interface polyhedronInputs {
+  vertices: Array<number>;
+  indices: Array<number>;
+  radius: number;
+  detailDivisor: number;
+}
+
+interface planeSpecs {
+  apiVersion: number;
+  id: number;
+  author: string;
+  created: string;
+  updated: string;
+  vertexGeometry?: polyhedronInputs;
+  basicGeometry?: basicGeometry;
+  controlLoops: Array<controlLoop>;
+}
+function genBasicPlane(): planeSpecs {
+  const basicGeometry = {
+    wingspan: 1.1,
+    length: 0.4,
+    chord: 0.1,
+  };
+  const interrupts = [toControls.yaw, toControls.pitch, toControls.roll];
+  const controlLoop = {
+    frequency: 600,
+    interrupts: interrupts,
+  };
+  const controlLoops = [controlLoop];
+  const plane = {
+    apiVersion: 0.1,
+    id: 0,
+    author: 'default',
+    created: '2024-02-27T05:27:00.000Z',
+    updated: '2024-02-27T05:27:00.000Z',
+    basicGeometry: basicGeometry,
+    controlLoops: controlLoops,
+  };
+  return plane;
+}
 
 export class Plane {
   private static readonly MATERIAL = new THREE.MeshNormalMaterial();
-  private wingspan: number;
-  private length: number;
-  private chord: number;
-  private model: THREE.Group;
-
-  constructor(wingspan: number, length: number, chord: number) {
-    this.wingspan = wingspan;
-    this.length = length;
-    this.chord = chord;
-    this.model = this.genStandinGeometry(wingspan, length, chord);
+  private static readonly defaultWingspan = 1.1;
+  private static readonly defaultLength = 0.4;
+  private static readonly defaultChord = 0.1;
+  model: THREE.Group;
+  private planeSpecs: planeSpecs;
+
+  constructor() {
+    this.planeSpecs = option.getOrElse(() => genBasicPlane())(
```

Comment:
I don't like this use of getOrElse because it's hard to read -- `thing = getOrElse(defaultValue)(expectedValue)` reads out of order (we don't "get the default value, else expected"). If you want to use this functional programming library, I think this is a more readable way to do it:

```
this.planeSpecs = pipe(this.importPlane(JSON.stringify(defaultPlane)), getOrElse(() => genBasicPlane()));
```

Typescript/Javascript already supports a couple built in, easier-to-read ways to say "get this if it exists, otherwise this" -- https://www.typescriptlang.org/play#example/nullish-coalescing. In other languages, there are often builtin Optional types.

I don't see much functional programming at work outside of handling sets of data -- it's generally not worth it for a single object.